llvm/docs/AMDGPUUsage.rst

```diff
@@ -5969,6 +5969,31 @@ following sections:
 * :ref:`amdgpu-amdhsa-memory-model-gfx942`
 * :ref:`amdgpu-amdhsa-memory-model-gfx10-gfx11`
 
+.. _amdgpu-fence-as:
+
+Fence and Address Spaces
+++++++++++++++++++++++++++++++
+
+LLVM fences do not have address space information, thus, fence
+codegen usually needs to be conservative and fence all address spaces.
+
+In the case of OpenCL, where synchronization can only happen in the
+same address space, this can result in extra unnecessary waits.
```

Comment:
> can only happen in the same address space

Do you mean something more like "only needs to happen in specific address spaces"?

Reply:
Or to be more explicit, "only needs to happen in user-specified address spaces".